tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -113,6 +113,50 @@ TEST(KNN, kdtree_solver)
 	SG_UNREF(labels_test);
 }
 
+TEST(KNN, covertree_solver)
+{
+
+	int32_t num = 50;
+	int32_t feats = 2;
+	int32_t classes = 3;
+
```

Comment:
I suggest rather than copy pasting all this code, you add a `set_up` method that prepares all the data and instances, and then each unit test just chooses the solver type and checks output